packages/itwinui-css/src/table/table.scss

```diff
@@ -135,8 +135,6 @@
 }
 
 @mixin iui-table-row {
-  --_iui-table-cell-icon-fill: var(--iui-color-foreground-4);
```

Comment:
@mayank99 Removing L138 was not enough, unfortunately. It is giving a variable undefined error, as expected. 

![Screenshot from 2022-10-18 09-54-49](https://user-images.githubusercontent.com/45748283/196461778-d03bd463-6897-41e7-831c-dd1b21b56b24.png)

But, the fill attribute is still considered. Thus it still overrides the fill on the `<svg>` tag. In this example, I removed the fill from the path of the pdf icon and added a pink fill to the pdf `<svg>` tag. It still shows as black.

![Screenshot from 2022-10-18 09-57-35](https://user-images.githubusercontent.com/45748283/196462141-7fac6a40-bd27-479e-94e7-b2474d63977f.png)

Thus, I also changed [L452-465](https://github.com/iTwin/iTwinUI/pull/785/files#diff-3ac545d94bb9609fcd00db7227644a76d684ddad2b678eeaa75ac0d0209a251cL452-R465)

Reply:
can you try one more thing? set the fill on the wrapping element instead of the svg selector (it should get inherited by the svg)

I think that will allow the `fill` attribute on the `<svg>` to take preference, while still avoiding the default black fill that happens when no fill is specified.

Reply:
i think this means you should be able to revert most of the changes, right? the final diff should just have line 452 moved up a couple lines.

![](https://user-images.githubusercontent.com/9084735/196525215-241cc034-2ae6-4d15-9204-64a900ba866e.png)

Reply:
I tried it before and also now. I think the changes are needed or make the code cleaner.
* If I remove [L465-467](https://github.com/iTwin/iTwinUI/pull/785/files#diff-3ac545d94bb9609fcd00db7227644a76d684ddad2b678eeaa75ac0d0209a251cR465-R467), then the status icons become grey
![image](https://user-images.githubusercontent.com/45748283/196530826-c4200531-39eb-4ad1-843c-ea4145edb651.png)

* And then I removed [L138](https://github.com/iTwin/iTwinUI/pull/785/files#diff-3ac545d94bb9609fcd00db7227644a76d684ddad2b678eeaa75ac0d0209a251cL138), since you suggested that, also because now `fill` on [L441](https://github.com/iTwin/iTwinUI/pull/785/files#diff-3ac545d94bb9609fcd00db7227644a76d684ddad2b678eeaa75ac0d0209a251cR441), directly uses the color and not the variable. i.e. the variable is no longer used for the start icon fill

Reply:
Are you sure? I undid all your changes and just have this one line diff. It works fine
![](https://user-images.githubusercontent.com/9084735/196547675-6a490456-43c6-4ccd-a6f4-a7bde07e4db8.png)

Reply:
Ohh, thanks for confirming. You're right it works. I accidentally was using `fill: var(--iui-color-foreground-4);` on L449 instead of `var(--_iui-table-cell-icon-fill)`.

Updated the code to just include your above mentioned change.

Reply:
Actually, with the new code, looks like two tests failed: iTwinUI_table_Type_Expandable_Rows_0_demo-expandable-rows_0_desktop.png (Light and Dark theme)

![image](https://user-images.githubusercontent.com/45748283/196550868-80283d20-672c-4c14-b83e-1329648f077a.png)

Here the status icons are black instead of their status colors. This gets fixed with the code before the latest commit.

This is because in the latest commit, we only pass a fill as a property to the parent that will be inherited by the svg (very low specificity). `.iui-button-icon`'s black fill (`currentColor`) easily overrides it.

![MicrosoftTeams-image (3)](https://user-images.githubusercontent.com/45748283/196553712-176a7d83-626d-47f9-bb63-5b1ef02338a0.png)

In the previous commit, we pass the fill specifically on the svg of the end icon (higher specificity). This overrides the black fill of `.iui-button-icon`.

![MicrosoftTeams-image (2)](https://user-images.githubusercontent.com/45748283/196553702-50fefe77-bedd-4543-a866-b536c6e963dd.png)

<!-- ![MicrosoftTeams-image](https://user-images.githubusercontent.com/45748283/196552595-8dcac100-dd0e-41a0-9219-ab66fe3c0b8b.png)
![MicrosoftTeams-image (1)](https://user-images.githubusercontent.com/45748283/196552565-38e5e80f-1b01-4b99-b207-a6e46fa54792.png) -->

Reply:
Is the `iui-button-icon` class necessary? There is no button there.

Reply:
It's fine to leave it as is